Refactor COM Registration Fixing Tool part 2: remove bugs, return Windows errors, improve UX (#12355)
nvaccess/nvda: user_docs/en/userGuide.md

### Run COM registration fixing tool... {#RunCOMRegistrationFixingTool}

Installing and uninstalling programs on a computer can, in certain cases, cause COM DLL files to get unregistered.

> As accessibility Interfaces such as IAccessible depend on correct COM DLL registrations, issues can appear if the correct registrations are missing.

Reviewer: ```suggestion
As accessibility interfaces such as IAccessible depend on correct COM DLL registrations, issues can appear if the correct registrations are missing.
```

This can happen, for example, after installing and uninstalling Adobe Reader, Math Player and other programs.
It can also happen after Windows updates, or during other events that access the registry.

The missing registrations can cause issues in browsers, desktop apps, Windows Explorer, the task bar and other interfaces.

Specifically, the following issues can be solved by running this tool:

* NVDA reports "unknown" or "pane", when navigating in browsers such as Firefox or Edge, mail programs such as Thunderbird, etc.
* NVDA fails to switch between focus mode and browse mode when you expect it to.
* NVDA is very slow when navigating in browsers while using browse mode.
* And possibly other issues.

> Because this tool corrects entries in the Windows registry, it requires administrative access to work.

Reviewer: ```suggestion
Because this tool corrects entries in the Windows registry, it requires administrative access to work, just like when installing a program.
```

If you have UAC (User Access Control) enabled, as most users do, you will need to follow whatever prompts are presented by UAC, to run the tool successfully.
This may include entering your administrative account's password.

> It will not be NVDA asking for this, but Windows itself, just as it would while installing a program.

Reviewer: I think this sentence is kind of misleading, NVDA is requesting UAC elevation

```suggestion
```

Author: @seanbudd The "asking for this" was referring to the password, not the whole UAC process, but I can see where confusion could enter in that wording.

### Run COM registration fixing tool... {#RunCOMRegistrationFixingTool}

Installing and uninstalling programs on a computer can, in certain cases, cause COM DLL files to get unregistered.

> As accessibility interfaces such as IAccessible depend on correct COM DLL registrations, issues can appear if the correct registrations are missing.

Author: ```suggestion
Because accessibility interfaces such as IAccessible depend on correct COM DLL registrations, irregular behavior can appear if the correct registrations are missing.
```

> Because accessibility interfaces such as IAccessible depend on correct COM DLL registrations, irregular behavior can appear if the correct registrations are missing.

Author: ```suggestion
Because accessibility interfaces such as IAccessible depend on correct COM DLL registrations, abnormal behavior can appear if the correct registrations are missing.
```
